common/txmgr/inmemory_store.go

```diff
@@ -98,7 +100,31 @@ func (ms *inMemoryStore[CHAIN_ID, ADDR, TX_HASH, BLOCK_HASH, R, SEQ, FEE]) Creat
 	txmgrtypes.Tx[CHAIN_ID, ADDR, TX_HASH, BLOCK_HASH, SEQ, FEE],
 	error,
 ) {
-	return txmgrtypes.Tx[CHAIN_ID, ADDR, TX_HASH, BLOCK_HASH, SEQ, FEE]{}, nil
+	tx := txmgrtypes.Tx[CHAIN_ID, ADDR, TX_HASH, BLOCK_HASH, SEQ, FEE]{}
+	if ms.chainID.String() != chainID.String() {
+		panic(fmt.Sprintf("create_transaction: invalid chain ID: %s", chainID))
+	}
+
+	ms.addressStatesLock.RLock()
+	defer ms.addressStatesLock.RUnlock()
+	as, ok := ms.addressStates[txRequest.FromAddress]
+	if !ok {
+		return tx, fmt.Errorf("create_transaction: %w: %q", ErrAddressNotFound, txRequest.FromAddress)
+	}
+
+	// Persist Transaction to persistent storage
+	tx, err := ms.persistentTxStore.CreateTransaction(ctx, txRequest, chainID)
+	if err != nil {
+		return tx, fmt.Errorf("create_transaction: %w", err)
+	}
+
+	// Update in memory store
+	// Add the request to the Unstarted channel to be processed by the Broadcaster
+	if err := as.addTxToUnstartedQueue(&tx); err != nil {
+		return *ms.deepCopyTx(tx), fmt.Errorf("create_transaction: %w", err)
```

Comment:
If adding the tx to the in-memory store fails, is it ok if the tx is still in the persistence store?

Reply:
Think we'd maybe just need to extract the idempotency key check and the capacity check so we can validate those at the beginning of this method. Then the `addTxToUnstartedQueue` method wouldn't need to return an error so we wouldn't have a consistency issue.